ocb3/src/util.rs

```diff
@@ -0,0 +1,44 @@
+use aead::generic_array::{typenum::U16, ArrayLength, GenericArray};
+
+const BLOCK_SIZE: usize = 16;
+pub(crate) type Block = GenericArray<u8, U16>;
+
+#[inline]
+pub(crate) fn inplace_xor<T, U>(a: &mut GenericArray<T, U>, b: &GenericArray<T, U>)
+where
+    U: ArrayLength<T>,
+    T: core::ops::BitXor<Output = T> + Copy,
+{
+    for (aa, bb) in a.as_mut_slice().iter_mut().zip(b.as_slice()) {
+        *aa = *aa ^ *bb;
+    }
+}
+
+/// Doubles a block, in GF(2^128).
+///
+/// Adapted from https://github.com/RustCrypto/universal-hashes/blob/9b0ac5d1/polyval/src/mulx.rs#L5-L18
+#[inline]
+pub(crate) fn double(block: &Block) -> Block {
+    let mut v = u128::from_be_bytes((*block).into());
+    let v_hi = v >> 127;
+
+    // If v_hi = 0, return (v << 1)
+    // If v_hi = 1, return (v << 1) xor (0b0...010000111)
+    v <<= 1;
+    v ^= v_hi ^ (v_hi << 1) ^ (v_hi << 2) ^ (v_hi << 7);
+    v.to_be_bytes().into()
+}
```

Comment:
I believe this polynomial is impl'd in the `dbl` crate, although I don't think we need to block a merge on that.

The reason `polyval` can't use that is because it's the reversed polynomial and little endian, whereas `dbl` is the big endian version.